bittensor/btlogging/loggingmachine.py

```diff
@@ -71,6 +71,7 @@ def __init__(self, config: bittensor.config, name: str = BITTENSOR_LOGGER_NAME):
         self._queue = mp.Queue(-1)
         self._name = name
         self._config = config
+        self._first_party_loggers = {name}
 
```

Comment:
I think, instead of having separate fields, change it to something like this:

```python
def __init__(self, config: bittensor.config, name: str = BITTENSOR_LOGGER_NAME):
    # basics
    super(LoggingMachine, self).__init__()
    self._queue = mp.Queue(-1)
    self._primary_loggers = {name}
    self._config = config

def register_primary_logger(name: str):
    self._primary_loggers.add(name)

def deregister_primary_logger(name: str):
    self._primary_loggers.remove(name)
```

this way the main `bittensor` logger isn't a special case that has to be handled separately.

Reply:
You'll need to make additional changes, there is another reference to `self._name` (line 229) that needs to be changed to reflect the use of `self._primary_loggers` instead, as described above